src/reactor.rs

```diff
@@ -465,57 +478,109 @@ impl Source {
 }
 
 /// Future for [`Async::readable`](crate::Async::readable).
-#[derive(Debug)]
 #[must_use = "futures do nothing unless you `.await` or poll them"]
-pub struct Readable(Ready);
+pub struct Readable<'a, T>(Ready<&'a crate::Async<T>, T>);
+
+impl<T> Future for Readable<'_, T> {
+    type Output = io::Result<()>;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        ready!(Pin::new(&mut self.0).poll(cx))?;
+        log::trace!("readable: fd={}", self.0.handle.source.raw);
+        Poll::Ready(Ok(()))
+    }
+}
+
+impl<T> fmt::Debug for Readable<'_, T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Readable").finish()
+    }
+}
+
+/// Future for [`Async::readable_owned`](crate::Async::readable_owned).
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct OwnedReadable<T>(Ready<Arc<crate::Async<T>>, T>);
```

Comment:
I find it a bit confusing it's `readable_owned` but `OwnedReadable`. Could this be changed to `ReadableOwned`?

Reply:
Renamed to ReadableOwned/WritableOwned.